source/run/src/TG4PrimaryGeneratorAction.cxx

```diff
@@ -59,128 +66,220 @@ TG4PrimaryGeneratorAction::~TG4PrimaryGeneratorAction()
 // private methods
 //
 
+//_____________________________________________________________________________
+G4bool TG4PrimaryGeneratorAction::CheckVMCStack(TVirtualMCStack* mcStack) const
+{
+
+  if (!mcStack) {
+    TG4Globals::Warning(
+      "TG4PrimaryGeneratorAction", "CheckVMCStack", "Found nullptr for stack.");
+    return false;
+  }
+
+  if (mcStack->GetNtrack() <= 0) {
+    TG4Globals::Warning("TG4PrimaryGeneratorAction", "CheckVMCStack",
+      "No primary particles found on the stack.");
+    return false;
+  }
+  return true;
+}
+
+//_____________________________________________________________________________
+G4bool TG4PrimaryGeneratorAction::CheckParticleDefinition(
+  const G4ParticleDefinition* particleDefinition,
+  const TParticle* particle) const
+{
+  if (!particleDefinition) {
+    TString text = "G4ParticleTable::FindParticle() failed for ";
```

Comment:
The text in warning does not correspond to the function,
there is no "G4ParticleTable::FindParticle()" call performed